torch/ao/quantization/backend_config/backend_config.py

```diff
@@ -63,32 +72,95 @@ class ObservationType(Enum):
     example: torch.cat, maxpool
     """
 
+
+@dataclass
+class DTypeWithConstraints:
+    """
+    Config for specifying additional constraints for a given dtype, such as quantization value
+    ranges and scale value ranges, to be used in :class:`~torch.ao.quantization.backend_config.DTypeConfig`.
+    """
+    dtype: torch.dtype
+    quant_min: Optional[float] = None
+    quant_max: Optional[float] = None
+    scale_min: Optional[float] = None
+    scale_max: Optional[float] = None
+
+
 @dataclass
 class DTypeConfig:
     """
     Config for the set of supported input/output activation, weight, and bias data types for the
     patterns defined in :class:`~torch.ao.quantization.backend_config.BackendConfig`.
     """
-    input_dtype: Optional[torch.dtype] = None
-    output_dtype: Optional[torch.dtype] = None
-    weight_dtype: Optional[torch.dtype] = None
+    input_dtype: Optional[DTypeWithConstraints] = None
+    output_dtype: Optional[DTypeWithConstraints] = None
+    weight_dtype: Optional[DTypeWithConstraints] = None
     bias_dtype: Optional[torch.dtype] = None
     is_dynamic: Optional[bool] = None
 
+    def __init__(
+        self,
+        input_dtype: Union[torch.dtype, DTypeWithConstraints, None] = None,
+        output_dtype: Union[torch.dtype, DTypeWithConstraints, None] = None,
+        weight_dtype: Union[torch.dtype, DTypeWithConstraints, None] = None,
+        bias_dtype: Optional[torch.dtype] = None,
+        is_dynamic: Optional[bool] = None,
+    ):
+        if isinstance(input_dtype, torch.dtype):
+            input_dtype = DTypeWithConstraints(dtype=input_dtype)
+        if isinstance(output_dtype, torch.dtype):
+            output_dtype = DTypeWithConstraints(dtype=output_dtype)
+        if isinstance(weight_dtype, torch.dtype):
+            weight_dtype = DTypeWithConstraints(dtype=weight_dtype)
+        self.input_dtype = input_dtype
+        self.output_dtype = output_dtype
+        self.weight_dtype = weight_dtype
+        self.bias_dtype = bias_dtype
+        self.is_dynamic = is_dynamic
+
+    def get_input_dtype(self) -> Optional[torch.dtype]:
+        """
+        Return the input dtype as a `torch.dtype`.
+        """
+        return self.input_dtype.dtype if self.input_dtype is not None else None
```

Comment:
if `dtype` in `DTypeWithConstraints` can be None, then we can just do `return self.input_dtype.dtype` or even remove this function

Reply:
Not sure if I understand, since `self.input_dtype` can still be None, so `self.input_dtype.dtype` would be a null pointer. Either way I think we still need this function

Reply:
commented in https://github.com/pytorch/pytorch/pull/85200/files#r978112822